server/src/internalClusterTest/java/org/opensearch/search/aggregations/bucket/TermsDocCountErrorIT.java

```diff
@@ -1106,6 +1106,11 @@ public void testDoubleValueFieldSubAggDesc() throws Exception {
      * 3 one-shard indices.
      */
     public void testFixedDocs() throws Exception {
+        // Force merge each shard down to 1 segment to verify results are the same between concurrent and non-concurrent search paths
+        // Else for concurrent segment search there will be additional error introduced during the slice level reduce and thus different
+        // buckets may be returned. See https://github.com/opensearch-project/OpenSearch/issues/11680",
+        client().admin().indices().prepareForceMerge("idx_fixed_docs_0", "idx_fixed_docs_1", "idx_fixed_docs_2").get();
```

Comment:
With force merging segments to 1, concurrent search path will have single slice.  Can we update the `shard_size` parameter instead to see if the returned results are same ?

Reply:
Without any additional modifications to this test we won't be able to make it work with just increasing the shard size. The assertions below ([for example](https://github.com/opensearch-project/OpenSearch/pull/11732/files/1c4c7ef68904120c7d4b1745ffc7d7808a46b235#diff-954b334a87adb1b7244d5ac449c329e9b5a274c8147c890f3dba3f8709400ad9R1126-R1131)), are checking for a specific doc count error which will change when we modify the shard size.

If we make the shard size > cardinality so we collect all the documents we can get rid of the doc count error and have the assertions work for both concurrent and non-concurrent path but that seems to go against the spirit of this test since it is testing that the doc count error is calculated correctly.

Reply:
I wrote up a more detailed explanation for the different possible cases here: https://github.com/opensearch-project/OpenSearch/pull/11732#issuecomment-1883786431.